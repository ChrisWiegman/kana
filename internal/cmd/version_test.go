package cmd

import (
	"bytes"
	"os/exec"
	"testing"

	"github.com/gkampitakis/go-snaps/snaps"
)

func TestVersion(t *testing.T) {
	cmd := exec.Command("../../build/kana", "version")
	var out bytes.Buffer
	cmd.Stdout = &out
	err := cmd.Run()
	if err != nil {
		t.Fatalf("Unexpected error: %v", err)
	}

	t.Run("Test the version command for appropriate output", func(t *testing.T) {
		snaps.MatchSnapshot(t, out.String())
	})
}
