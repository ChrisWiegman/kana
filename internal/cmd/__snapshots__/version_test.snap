
[TestVersion/Test_the_version_command_for_appropriate_output - 1]
Version: 1.0.0
Build Time: 2024-03-16_10:50:11PM

---
